docs/dotnet/dotnet-signalr-metrics.md

```diff
@@ -0,0 +1,68 @@
+# Semantic Conventions for SignalR metrics emitted by ASP.NET Core
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for SignalR metrics emitted by .NET components and runtime.
+
+**Disclaimer:** These are initial .NET metric instruments available in .NET 8 but more may be added in the future.
+
+<!-- toc -->
+
+- [Metric: `signalr_http_transport.connection.duration`](#metric-signalr_http_transportconnectionduration)
+- [Metric: `signalr_http_transport.active_connections`](#metric-signalr_http_transportactive_connections)
+
+<!-- tocstop -->
+
+## Metric: `signalr_http_transport.connection.duration`
+
+**Status**: [Experimental][DocumentStatus]
+
+This metric is required.
+
+<!-- semconv metric.signalr.server.connection.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `signalr_http_transport.connection.duration` | Histogram | `s` | The duration of connections on the server. [1] |
```

Comment:
Porting active comment threads from the word doc:

## James Newton-King
I don't think "connections" should be pluralized here.

July 18, 2023 at 7:46 PM
## James Newton-King
Repeat for other categories, e.g. "request_leases.duration" -> "request_lease.duration"

July 18, 2023 at 7:47 PM
## Noah Falk
OTel's conventions seemed contradictory here, I'm hoping Liudmila can suggest what we should follow and what we should ignore:


Things that suggested plural:

- In the database metrics there are a bunch of metrics that use 'connections' plural. 

- In the pluralization example for count it suggested you would have a namespace 'system.processes' and presumably there is more than just the count metric in that namespace. I'm not sure if was proposing that only metrics that count processes go in there and there would be a separate system.process.* namespace for process related metrics that aren't counting processes.


Things that suggested singular:

- The pluralization rule said not to make it plural unless it was counting instances of a countable quantity and time isn't countable.


Things just felt odd:

- The pluralization rule seems to conflict with the desire to have consistent choice of pluralization when using categorization. For example a namespace like http.server.connection.* could have multiple metrics in it. Some metrics might count connections (e.g. http.server.connections.count) and others might measure a different unit like time (http.server.connection.duration). It feels odd if we wind up with two different categories, one singular and one plural depending on the unit of the final metric. 


Liudmila if you say which one is right I'm happy to follow it, but I guess also take this as feedback that the current rules and examples around pluralization didn't appear consistent.

July 19, 2023 at 4:23 AM
## Noah Falk
@Liudmila Molkova - Hoping to get your feedback on this one.

July 20, 2023 at 1:45 AM
## Trask Stalnaker
Trying to figure out the "rule" here, but connection.duration does sound better to me over connections.duration at least initially.

July 21, 2023 at 8:53 AM